apps/examples/secure_storage/secure_storage.c

```diff
@@ -0,0 +1,187 @@
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <security/security_common.h>
+#include <security/security_api.h>
+#include <tinyara/seclink.h>
+#include <tinyara/security_hal.h>
+#include <security/security_ss.h>
+#define SE_IOTIVITY_SVR_DB 0
+extern int rtl_ss_flash_read(uint32_t address, uint32_t len, uint8_t *data, int en_display);
+static char a[8092];
+static char b[8092];
+static int DATA_LENGTH = 128;
+
+#ifdef CONFIG_AMEBAD_TRUSTZONE
+static uint32_t SLOT0_START_ADDR = 0x1a000;
+static uint32_t SLOT1_START_ADDR = 0x1d000;
+#endif
+#ifdef CONFIG_AMEBALITE_TRUSTZONE
+static uint32_t SLOT0_START_ADDR = 0x24000;
+static uint32_t SLOT1_START_ADDR = 0x27000;
+#endif
+static uint32_t SLOT_SIZE = 0x1000;
+static uint32_t SLOT_AREA_OFFEST = 0x23000;
+
+/****************************************************************************
+ * secure_storage_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int sstorage_main(int argc, char *argv[])
+#endif
+{
+	printf("Start testing SE Secure Storage!!\n");
+	int length = DATA_LENGTH;
+	memset(a,'A', length);
+	a[length] = '\0';
+
+	printf("Start Security init!!\n");
+	sleep(1);
+	security_handle hnd;
+	security_error res = security_init(&hnd);
+	if (res != SECURITY_OK) {
+		dbg("security_init() Fail\n",__LINE__);
+		sleep(1);
+	}
+	printf("=====================%d==================================================\n",__LINE__);
+
+	sleep(1);
+	security_data data = { (void *)a, length};
+	char ss_path[7];
+
+	printf("Start Secure Storage write!!\n");
+	int slot;
+	for (slot = 0; slot < 4; slot++) {
```

Comment:
Please define and use it like `#define SS_SLOT_COUNT 4`

Reply:
Done.